tests/MODULE.bazel

```diff
@@ -1,6 +1,156 @@
-###############################################################################
-# Bazel now uses Bzlmod by default to manage external dependencies.
-# Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel.
-#
-# For more details, please check https://github.com/bazelbuild/bazel/issues/18958
-###############################################################################
+module(name = "rules_scala_annex_tests")
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "rules_java", version = "7.12.2")
+bazel_dep(name = "rules_jvm_external", version = "6.5")
+bazel_dep(name = "rules_scala_annex")
+local_path_override(
+    module_name = "rules_scala_annex",
+    path = "..",
+)
+
+register_toolchains(
+    "//:repository_default_toolchain_21_definition",
+    "//dependencies/indirect:test_zinc_2_13_direct_deps_off",
+    "//dependencies/indirect:test_zinc_2_13_used_deps_off",
+    "//plugins/kind-projector:test_zinc_2_13_with_kind_projector",
+    "//plugins/semanticdb:scala_2_13_with_semanticdb",
+    "//plugins/semanticdb:scala_3_with_semanticdb",
+    "//scala:test_bootstrap_2_12",
+    "//scala:test_bootstrap_2_13",
+    "//scala:test_bootstrap_3",
+    "//scala:test_zinc_2_12",
+    "//scala:test_zinc_2_13",
+    "//scala:test_zinc_3",
+    "//scala:test_zinc_2_13_fatal_deprecation_opt",
+    "//scala:test_zinc_2_13_log_level_debug",
+)
+
+# Please ensure these stay up-to-date with the versions in `rules/scala/versions.bzl`.
+# Unfortunately, `MODULE.bazel` files can't call `load`, so we have to copy them here.
+scala_2_12_version = "2.12.19"
+
+scala_2_13_version = "2.13.15"
+
+scala_3_version = "3.5.2"
+
+scalapb_version = "0.11.17"
+
+zinc_version = "1.10.4"
+
+specs2_version = "4.20.9"
+
+scalatest_version = "3.2.19"
+
+scalacheck_version = "1.18.1"
+
+annex_test = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
```

Comment:
I think for tests we may be able to use `include` because folks won't be depending on the test workspace: https://bazel.build/rules/lib/globals/module#include

Not sure if we should, but it is a thought.

Reply:
Unfortunately, we can't `include` an `annex.MODULE.bazel` file to use `@annex` in tests because, according to the [documentation](https://bazel.build/rules/lib/globals/module#include), "only files in the main repo may be included".

I considered symlinking an `tests/annex.MODULE.bazel` file to `annex.MODULE.bazel`, but then `rules_scala_annex` would have to `include` it as well. That doesn't work because in the context of the `rules_scala_annex`-`rules_scala_annex_test` relationship, `rules_scala_annex` is not the root module (it's a dependency), and "Only the root module may use `include()`; it is an error if a `bazel_dep`'s `MODULE` file uses `include()`".

Honestly, though, now that I think about it, `rules_scala_annex_test` using `@annex` was a bad idea to begin with for a couple reasons:
- It's probably best for testing if it uses `rules_scala_annex` like any other repository would, which means using a separate dependency tree
- Its dependencies are different from `rules_scala_annex`; this is what we have `@annex_test` for

Reply:
Fair. I like this argument.